coral-spark/src/main/java/com/linkedin/coral/spark/CoralSpark.java

```diff
@@ -115,6 +118,43 @@ private static String constructSparkSQL(RelNode sparkRelNode) {
     return rewrittenSparkSqlNode.toSqlString(SparkSqlDialect.INSTANCE).getSql();
   }
 
+  /**
+   * Users use this function as the main API for getting CoralSpark instance.
+   *
+   * Internally Appropriate parts of Sql RelNode is converted to Spark RelNode, Spark RelNode is converted back
+   * to SqlNode and SqlNode to SparkSQL.
+   *
+   * It returns an instance of CoralSpark which contains
+   *  1) Spark SQL
+   *  2) Base tables
+   *  3) Spark UDF information objects, ie. List of {@link SparkUDFInfo}
+   *
+   * @param sqlNode CoralNode which will be translated to SparkSQL.
+   * @param convertor Functional Interface to convert SqlNode to appropriate RelNode
+   *
+   *
+   * @return [[CoralSparkInfo]]
+   */
+  public static CoralSpark create(SqlNode sqlNode, Function<SqlNode, RelNode> convertor){
+    SparkRelInfo sparkRelInfo;
+    //apply RelNode transformations for sqlNode eligible for transformation.
+    if(sqlNode instanceof SqlCreate) {
```

Comment:
How about adding an interface `SqlCommand`, and have `SqlCreateTable` extend `SqlCreate` and implement `SqlCommand`? `SqlCommand` defines `getSelectQueries()` and `setSelectQueries()`. Those are just list versions of what you have now in `SqlCreateTable`.

Reply:
yeah, this sounds good

Reply:
I feel that `getSelectQueries()` and `setSelectQueries()` will be used for `CREATE TABLE/VIEW....AS`. If there's no other use case, can we rename `SqlCommand` -> `SqlCreateCommand`?

Reply:
Not sure, but I think commands like `INSERT OVERWRITE TABLE tmp  select * FROM foo`, might also leverage this